[6.1] Drop Bootstrap::framework

dgrammatiko opened the pull request:
Pull Request for Issue # .

### Summary of Changes
- B/C break
- Drop Bootstrap::framework Since J4 any bootstrap (js) component should be specifically loaded, the framework fn was kept for compatibility


### Testing Instructions

- code review
- Add `HTMLHelper::_('bootstrap.framework');` somewhere in the `index.php` of the atum template
- Check that all the BS components are registered 

![Screenshot 2025-02-24 at 11 43 22](https://github.com/user-attachments/assets/1e0c6fa2-96cd-4a5f-a076-1d7669b7facc)

### Actual result BEFORE applying this Pull Request



### Expected result AFTER applying this Pull Request



### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [x] Pull Request link for manual.joomla.org: [PR](https://github.com/joomla/Manual/pull/396)
- [ ] No documentation changes for manual.joomla.org needed

dgrammatiko committed:
Drop Bootstrap::framework

HLeithner reviewed (dismissed):
can you please do only on chance in one pr.

* you update package-lock.json (version numer)
* you change code style
* you use deprecated Factory::getDocument()

and the only thing this PR should do is to remove lines 588 to 616 in the Bootstrap.php

laoneo commented:
Istead fo removing, I would move it to the compat plugin.

dgrammatiko committed:
move the code to the compat plg

Hackwar commented:
Do we have to add the deprecated tag to the methods in the compat plugin as well?

dgrammatiko commented:
> Do we have to add the deprecated tag to the methods in the compat plugin as well?

To me it feels redundant, as it will never show up in the IDE

Hackwar commented:
I agree that it wont show up in the IDE, but it will prevent any discussions in the future about removal and deprecation if you move the tag from the old code also over to the new code.

dgrammatiko committed:
add back the deprecation

Hackwar commented:
Thank you!

HLeithner commented:
the implementation in the b/c plugin is not optimal.

please add a new Class for this function and also an option to disable "all html helper functions" (do we have more?) and make it as option please.

dgrammatiko commented:
> please add a new Class

what namespace? is `namespace Joomla\Plugin\Behaviour\Compat\HTML\Bootstrap` ok?

> and also an option

simple toggle should be ok here?

HLeithner commented:
yes namespace is ok, we already have yes/no fields in the config. something like that

dgrammatiko committed:
use own class, use toggler

dgrammatiko commented:
- toggler added
- moved to own class

![Screenshot 2025-02-24 at 19 24 54](https://github.com/user-attachments/assets/6edad436-ed36-46fe-9b78-583794a378fd)

dgrammatiko committed:
cs

dgrammatiko committed:
meh

fgsw commented on administrator/language/en-GB/plg_behaviour_compat.ini:
```diff
@@ -11,3 +11,5 @@ PLG_COMPAT_FIELD_ES5_ASSETS_LABEL="ES5 Assets"
 PLG_COMPAT_FIELD_REMOVED_ASSETS_DESCRIPTION="Activate this option if your extension requires removed assets which has resulted in an exception. The assets provided are empty but prevent the exception."
 PLG_COMPAT_FIELD_REMOVED_ASSETS_LABEL="Removed Assets"
 PLG_COMPAT_XML_DESCRIPTION="If you use extensions which are not using the current Joomla Coding standards then this plugin, when enabled, will provide backward compatibility to the prior major version."
+PLG_COMPAT_FIELD_HTML_HELPERS_LABEL="HTML Helpers"
+PLG_COMPAT_FIELD_HTML_HELPERS_DESCRIPTION="Activate this option if your extension requires deprecated HTMLHelper classes or functions. When enabled, will provide backward compatibility to the prior major version."
```
Should be moved up below:
PLG_COMPAT_FIELD_ES5_ASSETS_LABEL="ES5 Assets"

dgrammatiko replied:
done

Fedik commented on plugins/behaviour/compat/src/Extension/Compat.php:
```diff
@@ -92,6 +93,15 @@ public function __construct(DispatcherInterface $dispatcher, array $config = [])
+         */
+        if ($this->params->get('html_helpers', '1')) {
+            // Restore HTMLHelper::Bootstrap('framework')
+            new Bootstrap();
```
Use correct registry:
```php
Factory::getContainer()->get(\Joomla\CMS\HTML\Registry::class)
			->register('bootstrap', \Blabla\HTML\Bootstrap::class, true);
```

Then class `Blabla\HTML\Bootstrap` should extend existing `Joomla\CMS\HTML\Helpers\Bootstrap`.
And remove the constructor from `Blabla\HTML\Bootstrap`.

I think it need to be done onAfterRoute, or maybe even onAfterInitialise, because extensions may use that for override existing bootstrap helper. And we override their overrides. @HLeithner what do you think?

HLeithner replied:
yes should be done as earl as possible and the constructor makes no sense, call the registration directly in the plugin method.

dgrammatiko replied:
Done

dgrammatiko committed:
refactor

dgrammatiko committed:
cs

Fedik commented on plugins/behaviour/compat/src/HTMLHelper/Bootstrap.php:
```diff
@@ -0,0 +1,39 @@
+     *
+     * @return  void
+     *
+     * @deprecated  4.0 will be removed in 6.0
```
I think the message need to be removed, or maybe changed? no idea :)

dgrammatiko replied:
Hannes asked to include the deprecation as per the original fn

HLeithner replied:
it's ok to keep bit but change the removed in to 7.0 please

dgrammatiko replied:
done

dgrammatiko committed:
Update Bootstrap.php

Fedik commented:
I have tested this item :white_check_mark: successfully on 30d3736d59b0cd68d6e5a9a8c588adb81274c997<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44991">issues.joomla.org/tracker/joomla-cms/44991</a>.</sub>

brianteeman commented on administrator/language/en-GB/plg_behaviour_compat.ini:
```diff
@@ -8,6 +8,8 @@ PLG_COMPAT_FIELD_CLASSES_ALIASES_LABEL="Classes Aliases"
 PLG_COMPAT_FIELD_CLASSES_ALIASES_DESCRIPTION="Add class aliases for classes which have been renamed or moved to a namespace."
 PLG_COMPAT_FIELD_ES5_ASSETS_DESCRIPTION="Activate this option if your extension requires *.es5 assets which has resulted in an exception. The assets provided are empty but prevent the exception."
 PLG_COMPAT_FIELD_ES5_ASSETS_LABEL="ES5 Assets"
+PLG_COMPAT_FIELD_HTML_HELPERS_DESCRIPTION="Activate this option if your extension requires deprecated HTMLHelper classes or functions. When enabled, will provide backward compatibility to the prior major version."
```
why not state the version number

dgrammatiko replied:
I've tried the not original take here and I copied that part from here: 

```
PLG_COMPAT_XML_DESCRIPTION="If you use extensions which are not using the current Joomla Coding standards then this plugin, when enabled, will provide backward compatibility to the prior major version."
```

brianteeman commented on administrator/language/en-GB/plg_behaviour_compat.ini:
```diff
@@ -8,6 +8,8 @@ PLG_COMPAT_FIELD_CLASSES_ALIASES_LABEL="Classes Aliases"
 PLG_COMPAT_FIELD_ES5_ASSETS_DESCRIPTION="Activate this option if your extension requires *.es5 assets which has resulted in an exception. The assets provided are empty but prevent the exception."
 PLG_COMPAT_FIELD_ES5_ASSETS_LABEL="ES5 Assets"
+PLG_COMPAT_FIELD_HTML_HELPERS_DESCRIPTION="Activate this option if your extension requires deprecated HTMLHelper classes or functions. When enabled, will provide backward compatibility to the prior major version."
+PLG_COMPAT_FIELD_HTML_HELPERS_LABEL="HTML Helpers"
```
to avoid confusion should this not be specific and refer to boostrap html helpers

dgrammatiko replied:
Hmm, Harald suggested one switch for all (probably there are more, haven't really checked) of them: https://github.com/joomla/joomla-cms/pull/44991#issuecomment-2679041294

dgrammatiko replied:
FWIW one or multiple switches is fine by me

richard67 commented:
I've allowed myself to fix the conflicts in files "plugins/behaviour/compat/compat.xml" and "administrator/language/en-GB/plg_behaviour_compat.ini" which were caused by the recent merged of PR #44240 .

dgrammatiko committed:
fix conflicts

dgrammatiko commented:
@softforge ping

dgrammatiko committed:
meh

softforge commented:
Im getting ready for alpha3 but will take a look over the weekend. Im still seeing one issue form  @HLeithner , @HLeithner is it all good from your point of view?

dgrammatiko commented:
This should be ok now

dgrammatiko committed:
Update Compat6.php

dgrammatiko committed:
Update Compat6.php

dgrammatiko commented:
The class is now using the `onAfterInitialise` event as requested

dgrammatiko committed:
fixes

- use the dropdown module in the dropdown HTMLHelper
- Remove the call to `bootstrap.framework` in com_config as it was add in 4.0 alpha1 way before we switched to our own tabs script

QuyTon commented on plugins/behaviour/compat6/src/Extension/Compat6.php:
```diff
@@ -111,4 +113,27 @@ public function onAfterInitialiseDocument(AfterInitialiseDocumentEvent $event)
+
+    /**
+
+
+     * The after Initialise logic
```
```suggestion
    /**
     * The after Initialise logic
```

dgrammatiko committed:
Update plugins/behaviour/compat6/src/Extension/Compat6.php

Co-authored-by: Quy Ton <quy@nomonkeybiz.com>

QuyTon committed:
Merge branch '6.0-dev' into 6.0-dev/remove-bs-fw
